smart_lists/mixins.py

```diff
@@ -24,8 +27,40 @@ def get_queryset(self):
         if filters:
             for fltr in filters:
                 qs = qs.filter(**fltr)
+        qs = self.get_search_results(qs)
         return qs
 
+    def get_search_results(self, queryset):
+        """
+        borrowed from django-admin
+        @param queryset:
+        @return: filtered queryset
+        """
+        search_term = self.request.GET.get(self.search_query_parameter_name, None)
+
+        if search_term is None:
+            return queryset
+        # Apply keyword searches.
+        def construct_search(field_name):
+            if field_name.startswith('^'):
```

Comment:
i did take it from django-admin django/contrib/admin/options.py class ModelAdmin method get_search_result.
A crazy idea would be to call this method directly... (but that might break across django versions).
def get_search_results(self, request, queryset, search_term):

or i could rewrite the code using a more simple lookup with only contains, that would make the code much cleaner.